src/FluentValidation/AbstractValidator.cs

```diff
@@ -139,22 +132,18 @@ public abstract class AbstractValidator<T> : IValidator<T>, IEnumerable<IValidat
 		/// Note that cascade behaviour <i>between</i> rules is controlled by <see cref="ClassLevelCascadeMode"/>.
 		/// </para>
 		/// <para>
-		/// Do not set this to <see cref="CascadeMode.StopOnFirstFailure"/>.
-		/// <see cref="CascadeMode.StopOnFirstFailure"/> will be removed in a future release
-		/// and will throw an exception if used here.
+		/// This cannot be set to the deprecated <see cref="CascadeMode.StopOnFirstFailure"/>.
+		/// <see cref="CascadeMode.StopOnFirstFailure"/>. Attempting to do so it will actually
+		/// result in <see cref="CascadeMode.Stop"/> being used.
 		/// </para>
 		/// </summary>
 		public CascadeMode RuleLevelCascadeMode {
 			get => _ruleLevelCascadeMode();
-			set {
 #pragma warning disable 618
-				if (value == CascadeMode.StopOnFirstFailure) {
+			set => _ruleLevelCascadeMode = () => value == CascadeMode.StopOnFirstFailure

```

Comment:
@jsabrooke as above, I think the conversion should take place inside the setter rather than inside the delegate